arduino/libraries/librariesmanager/install.go

```diff
@@ -85,3 +91,99 @@ func (lm *LibrariesManager) Uninstall(lib *libraries.Library) error {
 	lm.Libraries[lib.Name].Remove(lib)
 	return nil
 }
+
+//InstallZipLib  installs a Zip library on the specified path.
+func (lm *LibrariesManager) InstallZipLib(libPath string) error {
+	libsDir := lm.getUserLibrariesDir()
+	if libsDir == nil {
+		return fmt.Errorf("User directory not set")
+	}
+	err := Unzip(libPath, libsDir.String())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//Unzip takes the ZipLibPath and Extracts it to LibsDir
+func Unzip(src string, dest string) error {
+
+	var filenames []string
+
+	r, err := zip.OpenReader(src)
+	if err != nil {
+		return err
+	}
+	defer r.Close()
+
+	for _, f := range r.File {
+
+		// Store filename/path for returning and using later on
+		fpath := filepath.Join(dest, f.Name)
+
+		// Check for ZipSlip. More Info: http://bit.ly/2MsjAWE
+		if !strings.HasPrefix(fpath, filepath.Clean(dest)+string(os.PathSeparator)) {
+			return fmt.Errorf("%s: illegal file path", fpath)
+		}
+
+		filenames = append(filenames, fpath)
+
+		if f.FileInfo().IsDir() {
+			// Make Folder
+			os.MkdirAll(fpath, os.ModePerm)
+			continue
+		}
+
+		// Make File
+		if err = os.MkdirAll(filepath.Dir(fpath), os.ModePerm); err != nil {
+			return err
+		}
+
+		outFile, err := os.OpenFile(fpath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, f.Mode())
+		if err != nil {
+			return err
+		}
+
+		rc, err := f.Open()
+		if err != nil {
+			return err
+		}
+
+		_, err = io.Copy(outFile, rc)
+
+		// Close the file without defer to close before next iteration of loop
+		outFile.Close()
+		rc.Close()
+
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+//InstallGitLib  installs a library hosted on a git repository on the specified path.
+func (lm *LibrariesManager) InstallGitLib(Name string, gitURL string) error {
+	libsDir := lm.getUserLibrariesDir()
+	if libsDir == nil {
+		return fmt.Errorf("User directory not set")
+	}
+	err := Fetch(Name, gitURL, libsDir.String())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//Fetch Clones the repository to LibsDir
+func Fetch(name string, url string, dest string) error {
```

Comment:
Much like `Unzip` this function can also be private, also the arguments type thing.